### ISSUE TITLE
How to connect to a remote server

### ISSUE THREAD

**Issue by bj97301** [adamhartford/SwiftR#19]
I am not having any luck receiving the connected callback.  I am sure its something I am doing.  Any ideas?
var persistentConnection: SignalR!
        
        persistentConnection = SwiftR.connect("http://10.3.29.79/gameserver/", connectionType: .Persistent) { connection in
            connection.received = { data in
                print(data!)
            }
            connection.connected = { print("connected: \(connection.connectionID)") }
            connection.connectionSlow = { print("connectionSlow") }
            connection.reconnecting = { print("reconnecting") }
            connection.reconnected = { print("reconnected") }
            connection.disconnected = { print("disconnected") }
        }
        
        // Send data
        persistentConnection.send("Persistent Connection Test")

        return true

**Comment by adamhartford** [comment]
Are you able to browse to http://10.3.29.79/gameserver/? You should see some SignalR error message. (You're definitely running on port 80?) Do you see a "disconnected" message instead?

It also looks like you're calling `persistentConnection.send()` immediately after `connect`. You'll want to make sure you don't start sending data until you're connected. You could move your send call to the connected callback, or use a button somewhere for testing, etc.

For example, to achieve what you're trying to do:

```swift
connection.connected = {
    print("connected!")
    connection.send("foo")
}
```

Also make sure you haven't declared your connection variable somewhere such that it falls out of scope. If you declare it in a method that calls and returns before everything happens, it's lost.

**Comment by bj97301** [comment]
Good feed back.  I am now seeing this error:
Error: Optional({
    column = 4521;
    errorMessage = "You are using a version of the client that isn't compatible with the server. Client version 1.5, server version 1.4.";
    line = 8;
    sourceURL = "file:///Users/ckknight/Library/Developer/Xcode/DerivedData/gift-card-poker-fyyrfbilvyxpyndxvwnjvtjxmtdr/Build/Products/Debug-iphonesimulator/SwiftR.framework/jquery.signalR-2.2.0.min.js";
})

I am using 2.1.2 on my server
